src/index.js

```diff
@@ -213,16 +219,137 @@ module.exports = class IPLDResolver {
 
       pull(
         pull.values([nodeAndCID]),
-        this._putStream(callback)
+        this._putStream((err) => {
+          if (err) {
+            return callback(err)
+          }
+          callback(null, nodeAndCID.cid)
+        })
+      )
+    }
+  }
+
+  treeStream (cid, path, options) {
+    if (typeof path === 'object') {
+      options = path
+      path = undefined
+    }
+
+    options = options || {}
+
+    // non recursive
+    const p = pullPushable()
+
+    if (!options.recursive) {
+      const r = this.resolvers[cid.codec]
+
+      this.bs.get(cid, (err, block) => {
+        if (err) {
+          return p(err)
+        }
+
+        r.resolver.tree(block, (err, paths) => {
+          if (err) {
+            return p(err)
+          }
+          paths.forEach((path) => p.push(path))
+          p.end()
+        })
+      })
+    }
```

Comment:
this could be written as

```js
if (!options.recursive) {
  p = pullDefer.source()
  waterfall([
    (cb) => this.bs.get(cid, cb),
    (block, cb) => r.resolver.tree(block, cb)
  ], (err, paths) => {
    if (err) {
      return p.abort(err)
    }
    p.resolve(pull.values(paths)
  })
}
```